gnovm/pkg/gnolang/preprocess.go

```diff
@@ -1857,7 +1857,10 @@ func Preprocess(store Store, ctx BlockNode, n Node) Node {
 				tvs := make([]TypedValue, numNames)
 				if numNames > 1 && len(n.Values) == 1 {
 					// special case if `var a, b, c T? = f()` form.
-					cx := n.Values[0].(*CallExpr)
+					cx, ok := n.Values[0].(*CallExpr)
+					if !ok {
+						panic("should not happen")
```

Comment:
Don't add any new "should not happen" panics, please.

Reply:
I notice many occurrences of "should not happen" in the code. I assumed it was common practice.

Indeed, it would be beneficial to include the reasons for the errors. What do you think ?

Reply:
Yeah, I'm pushing everyone working on the VM to stop adding these. They are a favourite of jae, but really the amount of times that I've had a "should not happen" panic shown in my face goes to show how much these should be avoided.

Reply:
Please make this error more descriptive; if you find others you want to improve, PRs are welcome

Reply:
Improved the error here: https://github.com/gnolang/gno/pull/2206/commits/bd4b92444e2494f21ab1c74c8d6b697d0ad6c755